src/async_impl/request.rs

```diff
@@ -87,21 +89,23 @@ impl Request {
 
 impl RequestBuilder {
     /// Add a `Header` to this Request.
-    pub fn header<H>(&mut self, header: H) -> &mut RequestBuilder
+    pub fn header<K>(&mut self, key: K, value: ::http::header::HeaderValue) -> &mut RequestBuilder
```

Comment:
This could probably match the [builder method in `http`](https://docs.rs/http/0.1.6/http/request/struct.Builder.html#method.header) which would greatly ease the user.